pandas/core/indexes/datetimes.py

```diff
@@ -552,31 +552,24 @@ def _parsed_string_to_bounds(self, reso: Resolution, parsed: datetime):
         end = self._maybe_cast_for_get_loc(end)
         return start, end
 
+    def _disallow_mismatched_indexing(self, key, one_way: bool = False) -> None:
+        """
+        Check for mismatched-tzawareness indexing and re-raise as KeyError.
+        """
+        try:
+            self._deprecate_mismatched_indexing(key, one_way=one_way)
+        except TypeError as err:
+            raise KeyError(key) from err
+
     def _deprecate_mismatched_indexing(self, key, one_way: bool = False) -> None:
```

Comment:
Is this function still used elsewhere and cannot be rolled into `_disallow_mismatched_indexing`?

Reply:
its still used in `_maybe_cast_slice_bound`, where I think TypeError makes more sense than KeyError